src/supplemental/nanolib/log_test.c

```
@@ -0,0 +1,323 @@
+	NUTS_TRUE(fp != NULL);
+
+	NUTS_PASS(nng_mtx_alloc(&mtx));
+	NUTS_PASS(log_add_fp(fp, NNG_LOG_INFO, mtx, &config));
+
+#ifdef ENABLE_LOG
+	log_set_level(NNG_LOG_INFO);
+	log_info("Test log to file");
+#endif
+
+	log_clear_callback();
+	fclose(fp);
```

Cppcheck (2.20.0)

[warning] If resource allocation fails, then there is a possible null pointer dereference

(nullPointerOutOfResources)